tests/Database/DatabaseEloquentModelTest.php

```diff
@@ -533,6 +534,35 @@ public function testHasCastsOnEnumAttribute()
         $this->assertTrue($model->hasCast('enumAttribute', StringStatus::class));
     }
 
+    public function testHasCastsOnBcmathNumber()
+    {
+        if (! $this->isBcMathNumberSupported()) {
+            $this->markTestSkipped('Requires BcMath\Number class.');
+        }
```

Comment:
Alternatively, you could use with attributes:
```suggestion
    #[RequiresPhp('^8.4')]
    public function testHasCastsOnBcmathNumber()
    {
```
or
```suggestion
    #[RequiresPhpExtension('bcmath')]
    public function testHasCastsOnBcmathNumber()
    {
```
or
```suggestion
    #[RequiresMethod(Number::class, '__construct')]
    public function testHasCastsOnBcmathNumber()
    {
```

Reply:
Thank you for letting me know something I didn't know. That's a good way to write it!

I considered the possibility of testing using polyfills, so I decided to check with `RequiresMethod`.

Reply:
You're welcome—always happy to introduce someone to something new 👍🏻